### DISCUSSION TITLE
Trigger on completion of two disjoint actions

### DISCUSSION THREAD

**Discussion by keerthanvasist** [community/community#25518] [Actions]
<p>I have a use case where I need to trigger an action upon the completion of BOTH of two upstream actions. </p>
<p>I know we could use <code>repository_dispatch</code> to trigger an action when we need to. But, suppose there are two tasks that are triggered together, and both take indeterminable amount of time to reach completion, and I need to trigger a third task after both of the tasks are completed. Is there a way to do that?</p>

**Comment by BrightRan** [comment, marked as answer]
<p>Suppose your three workflows are <strong>wf1</strong> , <strong>wf2</strong> and <strong>wf3</strong> , wf1 and wf2 are the upstream works triggered simultaneously. And you want that wf3 is triggered only when both wf1 and wf2 are completed.</p>
<p>As you mentioned, because wf1 and wf2 are triggered simultaneously, and the time they spend is indeterminable, so we can’t be sure which one of wf1 and wf2 we should add the <strong>repository_dispatch event</strong> into. And there is no any other method (APIs or actions) that can be used to monitor the status of wf1 and wf2 in real time.</p>
<p>However, if wf1 and wf2 are in the same repository, as a workaround, you can combine them as two separate jobs ( <strong>job1</strong> and <strong>job2</strong> ) in one workflow, and the two jobs can run in parallel. In the workflow, you also need a third job ( <strong>job3</strong> ) used to trigger wf3 via the repository_dispatch event after both job1 and job2 are completed.</p>
<pre><code>jobs:
  job1:
    name: Job for wf1
    ...
  job2:
    name: Job for wf2
    ...
  job3:
    name: Trigger wf3 via repository_dispatch
    needs: [job1, job2]
</code></pre>
<p>If wf3 is also in the same repository, you can combine wf3 into the workflow as <strong>job3</strong> instead of using the repository_dispatch event to trigger wf3.</p>
<pre><code>jobs:
  job1:
    name: Job for wf1
    ...
  job2:
    name: Job for wf2
    ...
  job3:
    name: Job for wf3
    needs: [job1, job2]
</code></pre>
